datastore/bin/upload_picture_set.py

```diff
@@ -0,0 +1,105 @@
+        connection_string (str): URL to connect to the Azure storage account
+        pictures (str): list picture in a string format
+        user_id (str): uuid of the user
+        seed_name (str): name of the seed
+        zoom_level (float): zoom level of the picture
+        nb_seeds (int): number of seeds in the picture
+    """
+    try:
+        if not seed.is_seed_registered(cursor=cursor, seed_name=seed_name):
+            raise seed.SeedNotFoundError(
+                f"Seed not found based on the given name: {seed_name}"
+            )
+        seed_id = seed.get_seed_id(cursor=cursor, seed_name=seed_name)
+
+        if not user.is_a_user_id(cursor=cursor, user_id=user_id):
+            raise user.UserNotFoundError(
+                f"User not found based on the given id: {user_id}"
+            )
+
+        picture_set = picture_set_metadata.build_picture_set(user_id, len(pictures))
+        picture_set_id = picture_query.new_picture_set(
+            cursor=cursor, picture_set=picture_set, user_id=user_id
+        )
+
+        folder_created = asyncio.run(
+            blob.create_folder(container_client, str(picture_set_id))
+        )
+        if not folder_created:
+            raise AlreadyExistingFolderError(f"Folder already exists: {picture_set_id}")
+        folder_url = container_client.url + "/" + str(picture_set_id)
+        picture_uploaded = []
+        for picture_encoded in pictures:
+
+            # Create picture instance in DB
+            empty_picture = json.dumps([])
+            picture_id = picture_query.new_picture(
+                cursor=cursor,
+                picture=empty_picture,
+                picture_set_id=picture_set_id,
+                seed_id=seed_id,
+            )
+            # Upload picture to Azure and retrieve link
+            asyncio.run(
+                blob.upload_image(
+                    container_client, picture_set_id, picture_encoded, picture_id
+                )
+            )
+            picture_link = folder_url + "/" + str(picture_id)
+            picture_uploaded.append(picture_link)
+            # Create picture metadata and update DB instance (with link to Azure blob)
+            picture = picture_metadata.build_picture(
+                pic_encoded=picture_encoded,
+                link=picture_link,
+                nb_seeds=nb_seeds,
+                zoom=zoom_level,
+                description="upload_picture_set script",
+            )
+            picture_query.update_picture_metadata(
+                cursor=cursor, picture_id=picture_id, metadata=picture
+            )
+
+        return picture_set_id
+    except Exception:
+        if folder_url is not None:
+            arg = f""""picture_set_uuid":'{picture_set_id}'"""
+            blobs = asyncio.run(blob.get_blobs_from_tag(container_client, arg))
+            container_client.delete_blobs(blobs)
+        raise UploadError("An error occured during the upload of the picture set")
```

Comment:
In another commit I have all the Exception type from this file being catched with an explanation (user issue, seed issue). This Exception is being held in case something unexpected happens (EX: DB connection issue) and not really a user error.